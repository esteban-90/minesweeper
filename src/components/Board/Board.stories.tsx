import type { Story, Meta } from '@storybook/react'
import { Board, type BoardProps } from './Board'

export default {
  title: 'Components/Board',
  component: Board,
} as Meta

const Template: Story<BoardProps> = (args) => <Board {...args} />

export const BoardExample = Template.bind({})

BoardExample.args = {
  time: '000',
  levels: ['beginner', 'intermediate', 'expert'],
  mines: '010',
}
