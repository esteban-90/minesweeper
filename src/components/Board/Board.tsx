import type { FC } from 'react'
import styled from '@emotion/styled'
import { Counter } from './Counter'
import { Level } from './Level'
import { Reset } from './Reset'

export interface BoardProps {
  /**
   * Timer
   */
  time: string
  /**
   * Possible game scenarios
   */
  levels: string[]
  /**
   * Action handler when the game reset button is called
   */
  reset: () => void
  /**
   * Bombs in the field
   */
  mines: string
}

export const Board: FC<BoardProps> = ({ time, levels, reset, mines }): JSX.Element => {
  return (
    <Wrapper>
      <Counter>{time}</Counter>
      <Level>{levels}</Level>
      <Reset reset={reset} />
      <Counter>{mines}</Counter>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  width: max-content;
  padding-bottom: 2vw;
  justify-content: space-between;
`
