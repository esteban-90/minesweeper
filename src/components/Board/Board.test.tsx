import { render } from '@testing-library/react'
import { Board, type BoardProps } from './Board'

describe('Board test cases:', () => {
  it('should render correctly', () => {
    const props: BoardProps = {
      time: '000',
      levels: ['beginner', 'intermediate', 'expert'],
      mines: '010',
      reset: () => null,
    }

    const { asFragment } = render(<Board {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
