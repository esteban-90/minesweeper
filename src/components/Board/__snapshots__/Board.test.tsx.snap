// Vitest Snapshot v1

exports[`Board test cases: > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1gexv9m"
  >
    <div
      class="css-kj9346"
    >
      000
    </div>
    <select
      class="css-160ys9i"
    >
      <option
        class="css-1ycxzc0"
        value="beginner"
      >
        beginner
      </option>
      <option
        class="css-1ycxzc0"
        value="intermediate"
      >
        intermediate
      </option>
      <option
        class="css-1ycxzc0"
        value="expert"
      >
        expert
      </option>
    </select>
    <button
      class="css-mdm7pv"
    >
      ğŸ™‚
    </button>
    <div
      class="css-kj9346"
    >
      010
    </div>
  </div>
</DocumentFragment>
`;
